katharsis-rs/src/main/java/io/katharsis/rs/JsonApiResponseFilter.java

```diff
@@ -84,4 +89,19 @@ private boolean isResourceResponse(Object response) {
 		return singleResource || resourceList;
 	}
 
+	/**
+	 * Checks the media type of the response and the type of the response data to determine whether
+	 * the response should be wrapped with the necessary JSON API tags.
+	 * @param responseContext the context for the current response
+	 * @return <code>true</code>, if the media type is {@link JsonApiMediaType#APPLICATION_JSON_API} and the
+	 * 			response is not already in JSON API format,<br />
+	 * 			<code>false</code>, otherwise.
+	 */
+	private boolean wrapResponse(ContainerResponseContext responseContext) {
+		boolean mediaTypeJsonApi = JsonApiMediaType.APPLICATION_JSON_API_TYPE.equals(responseContext.getMediaType());
+		Object response = responseContext.getEntity();
+		boolean alreadyJsonApi = response instanceof Document || response instanceof InputStream;
```

Comment:
Ah, cool. Could you make it into a map then of `Map<Class<?>,Boolean>` and just statically configure it? Or, a 
```
private static final Predicate<?> shouldWrap = 
Stream.<Predicate<Object>>of(InputStream.class::isInstance, Document.class::isInstance).reduce(Predicate::or).orElse(x -> false);

if(shouldWrap.test(responseContext.getEntity()))
   // ...
```
This means the fundamental code doesn't change and you can extend easily. Alternatively, you could have a `Map<Predicate<Object>,Function<Object,?>>` where in you iterate the entries and apply the first function that matches.